From 7f6f613826aef9810999c1599c8354e8a78fbdb4 Mon Sep 17 00:00:00 2001
From: Lukas1818 <44570204+Lukas1818@users.noreply.github.com>
Date: Fri, 19 Feb 2021 21:08:44 +0100
Subject: [PATCH] fixs for Phyton 3.9 (#2955)

* change deprecated isAlive() to is_alive()

* remove unsupported keyword "encoding"

Co-authored-by: Lukas1818 <git@lukas1818.de>
---
 horizons/util/loaders/jsondecoder.py | 2 +-
 horizons/util/preloader.py           | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/horizons/util/loaders/jsondecoder.py b/horizons/util/loaders/jsondecoder.py
index ea2ddfb330..138a948f42 100644
--- a/horizons/util/loaders/jsondecoder.py
+++ b/horizons/util/loaders/jsondecoder.py
@@ -37,4 +37,4 @@ def _decode_dict(dct):
 			return newdict
 
 		with open(path, "r") as f:
-			return json.load(f, encoding="ascii", object_hook=_decode_dict)
+			return json.load(f, object_hook=_decode_dict)
diff --git a/horizons/util/preloader.py b/horizons/util/preloader.py
index 5689cc6815..2e072fcc64 100644
--- a/horizons/util/preloader.py
+++ b/horizons/util/preloader.py
@@ -79,9 +79,9 @@ def wait_for_finish(self):
 		"""
 		self.lock.acquire()
 		# wait until it finished its current action
-		if self.isAlive():
+		if self.is_alive():
 			self.join()
-			assert not self.isAlive()
+			assert not self.is_alive()
 		else:
 			try:
 				self.lock.release()
